From 63fa7444e9c9a1f830079bf5df6603f9f4bfd8da Mon Sep 17 00:00:00 2001
From: WingMan Kwok <w-kwok2@ti.com>
Date: Thu, 22 Jun 2017 09:28:31 -0400
Subject: [PATCH 43/43] arm: dts: am57xx-idk-common: add support of hsr/prp
 vlan pcp to rx queue map"

Usage:

pcp-rxq-map = /bits/ 8 <3 3 2 2 1 1 0 0>;

where (from left to right) low-pcp to high-pcp:

pcp    rxq
-------------
0      3 (q4)
1      3 (q4)
2      2 (q3)
3      2 (q3)
4      1 (q2)
5      1 (q2)
6      0 (q1)
7      0 (q1)

Note 1:
If no pcp-rxq-map is defined, the default is
<3 3 2 2 1 1 0 0>.

Note 2:
If a pcp is mapped to a rx queue (even it is the
default case), it is user's responsibility to make
sure that the rx queue have enough buffers, ie. 48,
to receive at least one regular size frame.

Signed-off-by: WingMan Kwok <w-kwok2@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/boot/dts/am57xx-idk-common.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boot/dts/am57xx-idk-common.dtsi b/arch/arm/boot/dts/am57xx-idk-common.dtsi
index 408d7cfd..d0b9e09 100644
--- a/arch/arm/boot/dts/am57xx-idk-common.dtsi
+++ b/arch/arm/boot/dts/am57xx-idk-common.dtsi
@@ -165,6 +165,8 @@
 		interrupt-parent = <&pruss2_intc>;
 		/*rx-queue-size = /bits/ 16 <254 134 134 254 48>;*/ /* RED */
 		/*rx-queue-size = /bits/ 16 <194 194 194 194 48>;*/ /* EMAC */
+		/* map is lowpcp-to->hipcp where val 3=q4, 2=q3, 1=q2, 0=q1 */
+		/*pcp-rxq-map = /bits/ 8 <3 3 2 2 1 1 0 0>;*/
 
 		pruss2_emac0: ethernet-mii0 {
 			phy-handle = <&pruss2_eth0_phy>;
-- 
1.9.1

